### PR TITLE
fix: sdk 400 ensure empty key is not accepted as encryption key for

### PR DESCRIPTION
**Pull request by itsdebs**
**Fixes** # (*issue*)
Null or empty encryption key is not valid anymore. In case of null or empty key, the database is no longer encrypted.
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] Version upgraded (project, README, gradle, podspec etc)
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: core/src/main/java/com/rudderstack/android/sdk/core/persistence/DefaultPersistenceProviderFactory.java [hunk 1]
```diff
@@ -47,10 +48,11 @@ public PersistenceProvider create(Application application) {
...
         }
         if (isEncrypted) {
-            if (encryptionKey == null) {
-                RudderLogger.logWarn("DBPersistentManager: isEncrypted is true but encryptionKey is null. Proceeding with null key");
+            if (TextUtils.isEmpty(encryptionKey)) {
+                RudderLogger.logWarn("DBPersistentManager: isEncrypted is true but encryptionKey is null or empty. Proceeding with unencrypted database");
+                isEncrypted = false;
             }
-            if (encryptedDbName == null) {
+            else if (encryptedDbName == null) {
```
Comment (1abhishekpandey) [inline core/src/main/java/com/rudderstack/android/sdk/core/persistence/DefaultPersistenceProviderFactory.java:55]:
```suggestion
            if (encryptedDbName == null) {
```
Reason: It doesn't matter `encryptionKey` is empty or not, `encryptedDbName` can still be `null`, right? So we should handle that irrespective of `encryptionKey`.

**Review by pallabmaiti — APPROVED** [review]

File: core/src/main/java/com/rudderstack/android/sdk/core/persistence/DefaultPersistenceProviderFactory.java (hunk 1 above)
Comment (itsdebs) [reply core/src/main/java/com/rudderstack/android/sdk/core/persistence/DefaultPersistenceProviderFactory.java:55]: If isEncrypted is false, we will be creating unencrypted database, so anyhow encrypted database name will not be required.

Comment (1abhishekpandey) [reply core/src/main/java/com/rudderstack/android/sdk/core/persistence/DefaultPersistenceProviderFactory.java:55]: Just try simulating the above scenario (where encryptionKey and encryptedDbName are null), and you'll get an exception, over [here](https://github.com/rudderlabs/rudder-sdk-android/blob/11e3ac22e3e9d22dd9ac65b443330914df4cf584/core/src/main/java/com/rudderstack/android/sdk/core/persistence/DefaultPersistenceProvider.java#L94).

Please consider enhancing the case when DB name is `null`. We should assigning default values in that case.

Comment (itsdebs) [reply core/src/main/java/com/rudderstack/android/sdk/core/persistence/DefaultPersistenceProviderFactory.java:55]: We should be having null checks

**Review by 1abhishekpandey — APPROVED** [review]

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**